public/src/components/channelManagement/campaigns/CampaignsForm.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState } from 'react';
+import { makeStyles, Theme } from '@material-ui/core';
+import CampaignsSidebar from './CampaignsSidebar';
+import CampaignsEditor from './CampaignsEditor';
+
+import {
+  fetchFrontendSettings,
+  FrontendSettingsType,
+  saveFrontendSettings,
+} from '../../../utils/requests';
+
+import withS3Data, { DataFromServer, InnerProps } from '../../../hocs/withS3Data';
+
+const useStyles = makeStyles(({ spacing, typography }: Theme) => ({
+  viewTextContainer: {
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'center',
+    alignItems: 'center',
+    marginTop: '-50px',
+  },
+  viewText: {
+    fontSize: typography.pxToRem(16),
+  },
+  body: {
+    display: 'flex',
+    overflow: 'hidden',
+    flexGrow: 1,
+    width: '100%',
+    height: '100%',
+  },
+  leftCol: {
+    height: '100%',
+    flexShrink: 0,
+    overflowY: 'auto',
+    background: 'white',
+    paddingTop: spacing(6),
+    paddingLeft: spacing(6),
+    paddingRight: spacing(6),
+  },
+  rightCol: {
+    flexGrow: 1,
+    display: 'flex',
+    justifyContent: 'center',
+  },
+}));
+
+export interface Campaign {
+  name: string;
+  nickname: string;
+  description?: string;
+}
+export type Campaigns = Campaign[];
+
+const CampaignsForm: React.FC<InnerProps<Campaigns>> = ({
+  data: campaigns,
+  setData: setCampaigns,
+  saveData: saveCampaigns,
+}: InnerProps<Campaigns>) => {
+  const classes = useStyles();
+
+  const createCampaign = (campaign: Campaign): void => {
+    setCampaigns([...campaigns, campaign]);
+    setNewCampaignCreated(true);
+    setSelectedCampaign(campaign);
+  };
+
+  const saveAllCampaigns = (): void => {
+    saveCampaigns();
+    setNewCampaignCreated(false);
+  };
+
+  // const saveCampaign = (campaign: Campaign): void => {
```

Comment:
Commented out for now. Save and archive (not delete!) functional will come once we make decisions around HOW we want to approach campaign edit/locking, and campaign archiving

Reply:
Removed code - we'll add in editing functionality in a future PR